### PR TITLE
Add LXIO driver support

### PR DESCRIPTION
**Pull request by concubidated**
I still don't understand why p1 and p2 light output is reversed when testing with pro/pro2

Perhaps there is something obviously wrong here, or maybe the pro series was just backward the entire time?

### CODE REVIEW

File: src/main/ptapi/io/piuio/lxio/lxio.c [hunk 1]
```diff
@@ -94,38 +94,6 @@ void convert_output_to_lxio()
 
   /* Menu Buttons */
 
-  // this should be player 2, but its player 1
```
Comment (voidderef) [inline src/main/ptapi/io/piuio/lxio/lxio.c]: That change is actually unrelated to the commit. Can you carve that out and sqash into your lxio commit?

File: src/main/io/util/usb.c [hunk 2]
```diff
@@ -99,20 +99,6 @@ void *io_usb_open(uint16_t vid, uint16_t pid, uint16_t config, uint16_t iface)
     return NULL;
   }
 
-  ret = libusb_set_configuration(dev, config);
```
Comment (voidderef) [inline src/main/io/util/usb.c:102]: That change is also not too obvious. Would be good to have that in a separate commit as well. This might still have a negative impact on anything that is untested, so better visibility is really good to have imo.

File: src/main/ptapi/io/piuio/lxio/lxio.c [hunk 3]
```diff
@@ -0,0 +1,296 @@
...
+
+  if (!lxio_drv_device_write(lxio_light_state_buffer, LXIO_MSG_SIZE)) {
+    return false;
+  }
+
+  if (!lxio_drv_device_read(lxio_btn_state_buffer, LXIO_MSG_SIZE)) {
+    return false;
+  }
+
+  // TODO: lxio contains an unsigned 16bit packet counter in the last two LSB
```
Comment (voidderef) [inline src/main/ptapi/io/piuio/lxio/lxio.c:158]: Want to leave that as an open TODO? Could also be left simply as a comment of something that is known but not implemented for given reasons.

File: src/main/ptapi/io/piuio/lxio/lxio.c [hunk 4]
```diff
@@ -0,0 +1,296 @@
...
+}
+
+void ptapi_io_piubtn_close(void)
+{
+  lxio_drv_device_close();
+}
+
+bool ptapi_io_piubtn_recv(void)
+{
+    ptapi_io_piuio_recv();
```
Comment (voidderef) [inline src/main/ptapi/io/piuio/lxio/lxio.c:230]: I don't think we should be driving the IO in two places. This can lead to very nasty side-effects if the IO for the buttons and pads are being threaded by different threads. IIRC that happens at least on Pro/Pro2.

I suggest removing that as pumping the events in the piuio loop should be good enough for accuracy for the buttons as well. Please leave a comment explaining this here.

Please also leave a comment on the empty `ptapi_io_piubtn_send` call.

File: src/main/ptapi/io/piuio/lxio/lxio.c [hunk 5]
```diff
@@ -0,0 +1,296 @@
...
+
+const char* ptapi_io_piubtn_ident(void)
+{
+    return "lxio";
+}
+
+bool ptapi_io_piubtn_open(void)
+{
+  memset(lxio_btn_state_buffer, 0, sizeof(lxio_btn_state_buffer));
+  return lxio_drv_device_open();
```
Comment (voidderef) [inline src/main/ptapi/io/piuio/lxio/lxio.c:220]: Same here regarding device management.

File: src/main/ptapi/io/piuio/lxio/lxio.c [hunk 6]
```diff
@@ -0,0 +1,296 @@
...
+
+bool ptapi_io_piubtn_send(void)
+{
+    return true;
+}
+
+void ptapi_io_piubtn_get_input(uint8_t player, struct ptapi_io_piubtn_inputs* inputs)
+{
+    // Set all inputs to off
+    if (player == 0) {
```
Comment (voidderef) [inline src/main/ptapi/io/piuio/lxio/lxio.c:242]: Hm, considering the threading situation mentioned, this part could run into consistency issues if the two IOs are threaded by separate threads. Locking can be quite expensive and would also need to be done in the `ptapi_io_piuio_recv` which is pumping the data. 

We can call it fine for now, but I would definitely leave a comment about the decision to "take the risk" now that there might be lost updates with the buttons. the pad inputs should not experience that if the IO is driven from there (which is the case right now).

File: src/main/ptapi/io/piuio/lxio/lxio.c (hunk 4 above)
Comment (concubidated) [reply src/main/ptapi/io/piuio/lxio/lxio.c:230]: I actually get a bunch of phantom inputs if I don't had this here as well.

File: src/main/ptapi/io/piuio/lxio/lxio.c (hunk 3 above)
Comment (concubidated) [reply src/main/ptapi/io/piuio/lxio/lxio.c]: Left it as a note

**Review by voidderef — APPROVED** [review]
